hooks/pre/31-mongo_storage_engine.rb

```diff
@@ -0,0 +1,53 @@
+pulp_db_param = param('katello', 'pulp_db_name') || param('foreman_proxy_content', 'pulp_db_name')
+fail_and_exit('Unknown pulp database name') unless pulp_db_param
+MONGO_DB_NAME = pulp_db_param.value
+
+def fail_and_exit(message)
+  Kafo::Helpers.log_and_say :error, message
+  kafo.class.exit 1
+end
+
+def migration
+  export_dir = '/var/tmp/engine_upgrade'
+  mongo_dir = '/var/lib/mongodb'
+  hiera_file = '/etc/foreman-installer/custom-hiera.yaml'
+  mongo_backup = '/var/tmp/mongo_backup'
+
+  # Create export directory and dump MongoDB
+  logger.info 'Stopping Pulp services except MongoDB'
+  Kafo::Helpers.execute('katello-service stop --exclude rh-mongodb34-mongod')
+  Dir.mkdir_p(export_dir)
+  File.chmod(0700, export_dir)
+  logger.info "Starting mongodump to #{export_dir}"
+  Kafo::Helpers.execute("mongodump --host localhost --out #{export_dir}")
+
+  # Move datafiles out of MongoDB directory so it will start
+  logger.info 'Export done, stopping MongoDB to remove old datafiles'
+  Kafo::Helpers.execute('systemctl stop rh-mongodb34-mongod')
+  logger.info "Moving contents from #{mongo_dir} to #{mongo_backup}"
+  Dir.mkdir_p(mongo_backup)
+  File.chmod(0700, mongo_backup)
+  Kafo::Helpers.execute("mv #{mongo_dir}/* #{mongo_backup}")
+
+  # Start MongoDB and import the dump, fail and notify user of backup if restore does not work.
+  logger.info 'Turning on MongoDB with wiredTiger and starting import'
+  Kafo::Helpers.execute('katello-service start --only rh-mongodb34-mongod')
+  Kafo::Helpers.execute("mongod --storageEngine wiredTiger --dbpath #{mongo_dir}")
+  Kafo::Helpers.execute("mongorestore --host localhost --db #{MONGO_DB_NAME} --quiet --dir #{export_dir}")
+  unless $?.success?
+    fail_and_exit "The restore could not be completed correctly, the backup is located at #{export_dir}"
```

Comment:
At this point, all services are still down right? Should we bring htem up back up or warn users?

Reply:
Also, given we've moved the mongodb data at this point, what should a user do?

Reply:
Updated error message on how to start services and to move the data back.